packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewer.js

```diff
@@ -85,10 +97,12 @@ class LightCurveViewer extends Component {
     const height = container.offsetHeight || 0
     const width = container.offsetWidth || 0
     this.drawChart(width, height, sameSubject)
+    this.updateInteractionMode(this.props.interactionMode)
```

Comment:
Should this always be called when the component updates? It seems you'd only want to do this if the `interactionMode` changes. Perhaps a check of `prevProps.interactionMode !== props.interactionMode` before calling?

I also have the same question about the `this.drawChart` method call above, even though line isn't being edited here.

Reply:
Adding the `(prevProps.interactionMode !== this.props.interactionMode)` for `updateInteractionMode()` makes sense 👌

I'm slightly wary to change the triggers to `componentDidUpdate()`'s `drawChart()` due to issues I previously encountered with the chart not updating when new annotations are added (which only appears in a later PR/branch)... but for the life of me I can't recall what those problems were.

So, for the moment, I've moved the `drawChart` trigger to the `if (!sameSubject)` body where it should _logically_ work, but I'll have to re-inspect this change once I dive back into my Add Annotations PR. 🤔 

Additional note: `LightCurveViewer.componentDidUpdate()` should only trigger in the following scenarios:
- New Subject is received.
- Change from Annotate mode to Move mode, or vice versa 
- New Annotation added
- Existing Annotation edited/deleted

Reply:
Extending `React.PureComponent` instead of `React.Component` can help here. Pure components have a simple `shouldComponentUpdate` method that only returns true if props or state have changed. It helps guard against updates running when you weren't expecting them.

Reply:
Noted, thanks Jim & Sarah. 👍  I'll have to see how the LCV can work as a PureComponent, but after I get the Add/Edit Annotations logic in.